docs/howto/weather-and-distance-tool/README.md

```diff
@@ -0,0 +1,411 @@
+* Initializes the dictionary it will return
+* Makes the GET request using the requests libray.
+* Checks that request's return code was successful.
+* Populates the JSON returned from the GET request to a Dictionary.
+* For each `dayKey`, gets the desired values from the response body and puts them in the return dictionary.
+
+```python
+    def _get_weather(self, destination, dayKeys: List[str]) -> dict:
+        ret = {}
+
+        params = {
+            "q": destination,
+            "days": 1,
+            "key": self.weather_key
+        }
+
+        r = requests.get(self.forecast_endpoint, params)
+
+        if r.status_code != 200:
+            self.provider.io.warn("_get_weather(%s) received error response %d" % (q, r.status_code))
+            for key in dayKeys:
+                ret[key] = None
+            return ret
+
+        json = r.json()
+
+        for key in dayKeys:
+            match = jp_ext.match('forecast.forecastday[0].day.%s' % key, json)
+            if (match and len(match) == 1):
+                ret[key] = match[0]
+
+        return ret
+```
+### Distance Matrix API Function
+
+The `_get_distance` function does the same steps as the `_get_chance_of_rain` function. The main difference is we set the return value to -1.0 and of course make the request to Google Maps.
+
+Also note that the distance values returned in this API are "8 mi", "12.5 mi", etc. So we use a regular expression to extract only the numeric value and convert it from a string to a float. 
+
+The code looks like this:
+
+```python
+    def _get_distance(self, destinationCity) -> float:
+        ret = -1.0
+
+        params = {
+            "destinations": destinationCity,
+            "origins": self.origin,
+            "units": self.units,
+            "key": self.distance_key
+        }
+        r = requests.get(self.distance_endpoint, params)
+        
+        if r.status_code != 200:
+            self.provider.io.error("get_distance received error response " + str(r.status_code))
+            return ret
+
+        json = r.json()
+        match = jp_ext.match('$.rows[0].elements.[0].distance.text', json)
+
+        if not match:
+            self.provider.io.info("no match")
+            return ret
+
+        if (len(match) == 1):
+            self.provider.io.info("match %s" % match[0])
+            re_match = re.search('(?P<dist>(.+)) mi', match[0])
+            if re_match:
+                ret = float(re_match.group("dist"))
+                self.provider.io.info("Got %f from %s" % (ret, match[0]))
+
+        return ret
+```
+
+### Data Processing
+Now it's time to get to the core of the plugin. We accept input that has a `City` column and use each city as a destination. We then call `_get_weather` and `_get_distance` for each city.
+
+We store an array of tuples consisting of each
+* Destination
+* Chance of Rain
+* Precipitation in inches
+* Minimum Temperature
+* Maximum Tempurature
+* Maximum Wind Speed
+* Distance from the Origin. 
```

Comment:
TW review: add colon in intro line, capitalize Inches, remove period from last item

We store an array of tuples consisting of each:
* Destination
* Chance of Rain
* Precipitation in Inches
* Minimum Temperature
* Maximum Tempurature
* Maximum Wind Speed
* Distance from the Origin